staging/src/k8s.io/apiserver/pkg/audit/request.go

```diff
@@ -42,6 +42,7 @@ func NewEventFromRequest(req *http.Request, level auditinternal.Level, attribs a
 		RequestReceivedTimestamp: metav1.NewMicroTime(time.Now()),
 		Verb:       attribs.GetVerb(),
 		RequestURI: req.URL.RequestURI(),
+		UserAgent:  req.UserAgent(),
```

Comment:
Mind adding a unit test? You could either add a unit test for this method, or check this in https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/apiserver/pkg/endpoints/audit_test.go

Reply:
we should also consider setting a maximum length on the size of the data allowed in this field

Reply:
Is there any doc saying that what value to set for user-agent? 
I only find apiserver set http.Server.MaxHeaderBytes=1 << 20 to limit max header size. Not find any thing about `User-Agent`

Reply:
Yes, I agree should limit size. My question is set this limit to what.

Reply:
xref: https://stackoverflow.com/questions/654921/how-big-can-a-user-agent-string-get

Reply:
How about 1024? And we should append something like `...TRUNCATED` if it's too long.

Reply:
works for me

Reply:
Thanks for your suggestion.